optapy-core/src/main/python/test/__init__.py

```diff
@@ -0,0 +1,609 @@
+            'PythonConstraintVerifier[Solution_]':
+        """
+        All subsequent calls to verify_that(constraint_function) will translate bytecode according to the rules
+        of the given BytecodeTranslation
+
+        :param bytecode_translation: A BytecodeTranslation member.
+        :return: self, for chaining
+        """
+        self.bytecode_translation = bytecode_translation
+        return self
+
+    @overload
+    def verify_that(self) -> 'PythonMultiConstraintVerification[Solution_]':
+        """
+        Creates a constraint verifier for all constraints of the ConstraintProvider.
+        """
+        ...
+
+    @overload
+    def verify_that(self, constraint_function: Callable[['ConstraintFactory'], 'Constraint']) -> \
+            'PythonSingleConstraintVerification[Solution_]':
+        """
+        Creates a constraint verifier for a given Constraint of the ConstraintProvider.
+        :param constraint_function: The constraint to verify
+        """
+        ...
+
+    def verify_that(self, constraint_function: Callable[['ConstraintFactory'], 'Constraint'] = None):
+        """
+        Creates a constraint verifier for a given Constraint of the ConstraintProvider.
+        :param constraint_function: Sometimes None, the constraint to verify. If not provided, all
+                                    constraints will be tested
+        """
+        return self.verifyThat(constraint_function)
+
+    @overload
+    def verifyThat(self) -> 'PythonMultiConstraintVerification[Solution_]':
+        """
+        Creates a constraint verifier for all constraints of the ConstraintProvider.
+        """
+        ...
+
+    @overload
+    def verifyThat(self, constraint_function: Callable[['ConstraintFactory'], 'Constraint']) -> \
+            'PythonSingleConstraintVerification[Solution_]':
+        """
+        Creates a constraint verifier for a given Constraint of the ConstraintProvider.
+        :param constraint_function: The constraint to verify
+        """
+        ...
+
+    def verifyThat(self, constraint_function: Callable[['ConstraintFactory'], 'Constraint'] = None):
+        """
+        Creates a constraint verifier for a given Constraint of the ConstraintProvider.
+        :param constraint_function: Sometimes None, the constraint to verify. If not provided, all
+                                    constraints will be tested
+        """
+        if constraint_function is None:
+            return PythonMultiConstraintVerification(self.delegate.verifyThat())
+        else:
+            return PythonSingleConstraintVerification(self.delegate.verifyThat(
+                PythonBiFunction(lambda _, constraint_factory:
+                                 constraint_function(PythonConstraintFactory(constraint_factory,
+                                                                             self.bytecode_translation)))))
+
+
+class PythonSingleConstraintVerification(Generic[Solution_]):
```

Comment:
Is it the intention for these classes to be visible to anyone?
What is the policy for this in Python, how does OptaPy specify what is API and what is impl?

Reply:
Ok, no visibility. This Python thing is starting to look like a scripting language.

Anyway... do we need the `Python` prefix in the name of these classes? Seems to me like bloat (obviously it's Python, in Python code). Also, it is camel case, which I understand that the Python people do not like.

Reply:
Classes names are normally PascalCase in Python (i.e. `SomeClass`, not `some_class`). Python Prefix can be removed.